nixos/lib/utils.nix

```diff
@@ -21,8 +21,8 @@ rec {
   toShellPath = shell:
     if types.shellPackage.check shell then
       "/run/current-system/sw${shell.shellPath}"
-    else if types.package.check shell then
-      throw "${shell} is not a shell package"
+    else if isString shell && types.path.check shell then
```

Comment:
Now that I think of it the proper way here might be to refine the type instead so that it accepts only string paths or `shellPackage`s.